### ISSUE TITLE
Support contributing demo example data via entry point

### ISSUE THREAD

**Issue by kitchoi** [enthought/traitsui#966]
After #908, the demo application will need to be able to load example files contributed by other projects.

This will involve creating an API for other projects to adhere to.
As a first cut, an entry point should return a path-like object for loading a directory of files and a display name for the data.

For future consideration, not part of this issue:
The API should be extensible so that options can be added.  Some ideas: we might want to extend the API so that projects can omit certain files under certain conditions at runtime, e.g. traitsui can choose to hide examples that require Wx in a Qt environment. There might be conditions that are settable by the users, e.g. traitsui can advertise a set of more mature examples and a set of less stable examples separately (see #949 as well), and the user might be allowed to select them?
